### PR TITLE
[WIP] Add dashboard and perf test for Azure under SIG Scalability

### PR DESCRIPTION
**Pull request by Jont828**

### CODE REVIEW

File: config/testgrids/kubernetes/sig-scalability/config.yaml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 dashboard_groups:
 - name: sig-scalability
   dashboard_names:
+    - sig-scalability-aks # TODO: name this aks or azure?
```
Comment (jackfrancis) [inline config/testgrids/kubernetes/sig-scalability/config.yaml:4]: We want this to be azure

**Review by BenTheElder — COMMENTED** [review]
Have you talked to sig scalability about this?

Also ... we're generally testing scalability _of kubernetes_ which we happen do on GCP and AWS because they provide $$$ resources so running it elsewhere isn't yet viable, but we're not in any way attempting to scale test the providers. As I understand it Azure is not making quite the same credit commitment and we should be careful about what we run on there as we finish eliminating any lingering company-internal accounts from upstream CI, because everything else Azure will need to fit in those credits.

cc @ameukam @upodroid @dims

**Comment by jackfrancis** [comment]
@BenTheElder thanks for jumping in

@mboersma and @Jont828 introduced this effort at a high level to SIG Scalability last week (ref: https://docs.google.com/document/d/1hEpf25qifVWztaeZPFmjNiJvPo-5JX1z0LSvvVY5G2g/edit#bookmark=kix.70akioyk9t2h) Agenda notes suggest that maybe the community infra aspect wasn't discussed. Sounds like we should keep SIG Testing in the loop on that and any ideas for future progress here? Happy to do so.

**Comment by BenTheElder** [comment]
> . Sounds like we should keep SIG Testing in the loop on that and any ideas for future progress here? Happy to do so.



More like SIG K8s Infra, and you don't have to loop us in, but ...  at the moment this just won't work as-written, since the build cluster selected doesn't have azure access _yet_, @jsturtevant is working with @upodroid on that part. You could switch to the google.com "default" cluster but ... we're phasing that out.

The part that may need discussing with k8s infra / Azure: It's not clear to me if this will fit, I don't know if the new budget has been disclosed yet (AFAIK there will not be an announcement from Microsoft, I think privately we have some idea) but at some point we'll be tracking under these updates https://kubernetes.slack.com/archives/CCK68P2Q2/p1719280357681539 where we will be posting the usage and remaining resources on the commuonity account.

... We'll want to make sure we can migrate all of the existing azure jobs or risk dropping them (https://groups.google.com/a/kubernetes.io/g/dev/c/p6PAML90ZOU)

So I'd personally suggest holding off making a large new Azure job until the effort to migrate the existing usage to community infra is complete. 

We don't plan to defer migrating Prow any longer, August 1 we will stop running unmigrated jobs. 
We've been attempting to migrate prow for years and have it ~almost there, and the engprod team running it will only be supporting through EOY so ... we kinda need to wrap this up and we're aiming to not port forward any dependency on any _other_ internal resources into the community clusters.
In the past / present this has included AWS, GCP, DO, Azure, VSphere, ... but it's _almost_ done and Azure in particular is being actively worked on.

I don't want to block anyone from making progress, but we're in a kind of awkward spot here and I think it probably makes sense to hold off for a little bit and/or see if you can help resolve the migration issues first.

**Comment by BenTheElder** [comment]
... reached out to clarify the budget, though the project also currently has zero visibility into the spend, so even once we know the budget we will have no idea how much capacity remains until everything switches over

**Comment by jackfrancis** [comment]
> So I'd personally suggest holding off making a large new Azure job until the effort to migrate the existing usage to community infra is complete.

This sounds reasonable to me!

We can make progress on our end w/ clusterloader2 as we ramp up towards having public testing on the new Azure infra.

**Comment by Jont828** [comment]
@BenTheElder Thanks for the feedback, I brought it up at the SIG scalability meeting and didn't realize there would be further considerations w.r.t. community infra. I'll go ahead and close the PR and get some more info about public testing on Azure infra.

**Pull request closed without merging**